src/sinks/gcp/cloud_storage.rs

```diff
@@ -0,0 +1,550 @@
+            creds.apply(&mut request);
+        }
+
+        let healthcheck =
+            self.client
+                .request(request)
+                .map_err(Into::into)
+                .and_then(healthcheck_response(
+                    self.creds.clone(),
+                    GcsError::BucketNotFound {
+                        bucket: self.bucket.clone(),
+                    }
+                    .into(),
+                ));
+
+        Ok(Box::new(healthcheck))
+    }
+}
+
+impl Service<RequestWrapper> for GcsSink {
+    type Response = hyper::Response<Body>;
+    type Error = hyper::Error;
+    type Future = hyper::client::ResponseFuture;
+
+    fn poll_ready(&mut self) -> Poll<(), Self::Error> {
+        Ok(().into())
+    }
+
+    fn call(&mut self, request: RequestWrapper) -> Self::Future {
+        let settings = request.settings;
+
+        let uri = format!("{}{}", self.base_url, request.key)
+            .parse::<Uri>()
+            .unwrap();
+        let mut builder = Request::builder();
+        builder.method(Method::PUT);
+        builder.uri(uri);
+        let headers = builder.headers_mut().unwrap();
+        headers.insert("content-type", settings.content_type);
+        headers.insert(
+            "content-length",
+            HeaderValue::from_str(&format!("{}", request.body.len())).unwrap(),
+        );
+        settings
+            .content_encoding
+            .map(|ce| headers.insert("content-encoding", ce));
+        headers.insert("x-goog-acl", settings.acl);
+        headers.insert("x-goog-storage-class", settings.storage_class);
+        for (p, v) in settings.metadata {
+            headers.insert(p, v);
+        }
+
+        let mut request = builder.body(Body::from(request.body)).unwrap();
+        if let Some(creds) = &self.creds {
+            creds.apply(&mut request);
+        }
+
+        self.client.request(request)
+    }
+}
+
+fn to_string(value: impl Serialize) -> String {
+    let value = serde_json::to_value(value).unwrap();
+    value.as_str().unwrap().into()
+}
+
+#[derive(Clone, Debug)]
+struct RequestWrapper {
+    body: Vec<u8>,
+    key: String,
+    settings: RequestSettings,
+}
+
+impl RequestWrapper {
+    fn new(req: PartitionInnerBuffer<Vec<u8>, Bytes>, settings: RequestSettings) -> Self {
+        let (body, key) = req.into_parts();
+
+        // TODO: pull the seconds from the last event
+        let filename = {
+            let seconds = Utc::now().format(&settings.time_format);
+
+            if settings.append_uuid {
+                let uuid = Uuid::new_v4();
+                format!("{}-{}", seconds, uuid.to_hyphenated())
+            } else {
+                seconds.to_string()
+            }
+        };
+
+        let key = format!(
+            "{}{}.{}",
+            String::from_utf8_lossy(&key[..]),
+            filename,
+            settings.extension
+        );
+
+        debug!(
+            message = "sending events.",
```

Comment:
I don't think this log event requires a period.

Reply:
Heh, I copied that from the `aws_s3.rs` sink. I'll change both.